crit/mempages.go

```diff
@@ -0,0 +1,141 @@
+package crit
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/checkpoint-restore/go-criu/v6/crit/images/pagemap"
+)
+
+var sysPageSize = uint64(syscall.Getpagesize())
+
+// MemoryReader is a struct used to retrieve
+// the content of memory associated with a specific process ID (pid).
+// New instances should be created with NewMemoryReader()
+type MemoryReader struct {
+	checkpointDir  string
+	pid            uint32
+	pagesID        uint32
+	pageSize       uint64
+	pagemapEntries []*pagemap.PagemapEntry
+}
+
+// NewMemoryReader creates a new instance of MemoryReader with all the fields populated
+func NewMemoryReader(checkpointDir string, pid uint32, pageSize uint64) (*MemoryReader, error) {
+	pSize := pageSize
+	// If the given page size is not a positive power of 2, set the default page size
+	if pageSize <= 0 || (pageSize&(pageSize-1)) != 0 {
```

Comment:
In the case where a user sets the page size to 0, it could be reasonable to fall back to a default value (i.e., sysPageSize). However, when a user specifies a page size that is not a positive power of 2, it would be more appropriate to return an error instead of silently overwriting the value.